website/static/js/pages/wiki-edit-page.js

```diff
@@ -1,16 +1,131 @@
 
-// Toggle tooltips
-$(function () {
-    $('[data-toggle="tooltip"]').tooltip();
-});
+var editable = (ctx.panelsUsed.indexOf('edit') !== -1);
+var viewable = (ctx.panelsUsed.indexOf('view') !== -1);
+var comparable = (ctx.panelsUsed.indexOf('compare') !== -1);
+var menuVisible = (ctx.panelsUsed.indexOf('menu') !== -1);
+
+var viewVersion = ctx.versionSettings.view || (editable ? 'preview' : 'current');
+var compareVersion = ctx.versionSettings.compare || 'previous';
+
+var wikiPageOptions = {
+    editVisible: editable,
+    viewVisible: viewable,
+    compareVisible: comparable,
+    menuVisible: menuVisible,
+    canEdit: ctx.canEdit,
+    viewVersion: viewVersion,
+    compareVersion: compareVersion,
+    urls: ctx.urls,
+    metadata: ctx.metadata
+};
+
+var wikiPage = new WikiPage('#wikiPageContext', wikiPageOptions);
+
+
+// Edit wiki page name
+if (ctx.canEditPageName) {
+    // Initialize editable wiki page name
+    var $pageName = $('#pageName');
+    $.fn.editable.defaults.mode = 'inline';
+    $pageName.editable({
+        type: 'text',
+        send: 'always',
+        url: ctx.urls.rename,
+        ajaxOptions: {
+            type: 'put',
+            contentType: 'application/json',
+            dataType: 'json'
+        },
+        validate: function(value) {
+            if($.trim(value) === ''){
+                return 'The wiki page name cannot be empty.';
+            } else if(value.length > 100){
+                return 'The wiki page name cannot be more than 100 characters.';
+            }
+        },
+        params: function(params) {
+            return JSON.stringify(params);
+        },
+        success: function(response, value) {
+            window.location.href = ctx.urls.base + encodeURIComponent(value) + '/';
+        },
+        error: function(response) {
+            var msg = response.responseJSON.message_long;
+            if (msg) {
+                return msg;
+            } else {
+                // Log unexpected error with Raven
+                Raven.captureMessage('Error in renaming wiki', {
+                    url: ctx.urls.rename,
+                    responseText: response.responseText,
+                    statusText: response.statusText
+                });
+                return 'An unexpected error occurred. Please try again.';
+            }
+        }
+    });
+}
 
-ShareJSDoc('.wiki', url, metadata);
\ No newline at end of file
+// Apply panels
+$(document).ready(function () {
+    var bodyElement = $('body');
+
+    $('*[data-osf-panel]').osfPanel({
+        buttonElement : '.switch',
+        onSize : 'xs',
+        'onclick' : function (title, thisbtn, event ) {
+            // this = all the column elements; an array
+            // title = Text of the button
+            // thisbtn = $(this);
+            // thisbtn.hasClass('btn-primary')
+
+            bodyElement.trigger('togglePanel', [
+                title.toLowerCase(),
+                thisbtn.hasClass('btn-primary')
```

Comment:
Not sure where my previous comment went, but this is still brittle. We can defer changing this, but it would be better if could get the toggle state of the clicked button in a more robust way, either by exposing `onEnable/onDisable` callbacks and or a single `onToggle` callback which takes `enabled` as an argument. ping @caneruguz